library/src/scala/NamedTuple.scala

```diff
@@ -209,6 +210,14 @@ object NamedTupleDecomposition:
     /** An immutable array consisting of all element values */
     inline def toIArray: IArray[Object] = x.toTuple.toIArray
 
+    /** An immutable map consisting of all element values.
+     *  Keys are the names of the elements.
+     */
+    inline def toMap: collection.immutable.Map[String, Tuple.Union[V]] =
+      summonAll[Tuple.Map[N, ValueOf]].toList
+        .map(_.asInstanceOf[ValueOf[? <: String]].value)
+        .lazyZip(x.toList)
+        .toMap
```

Comment:
You might also enjoy this `asInstanceOf`-less version of this code:

```scala
inline def toMap: collection.immutable.Map[String, Tuple.Union[V]] = {
    val names = inline compiletime.constValueTuple[N].toList match { case names: List[String] => names }
    names.lazyZip(x.toTuple.toList).toMap
  }
```

IMO it'd set some good precedent for people figuring stuff out around `inline` to start employing tricks like these.